examples/server_fns_axum/src/app.rs

```diff
@@ -945,3 +952,228 @@ pub fn PostcardExample() -> impl IntoView {
+}
+
+#[component]
+pub fn SimpleGenericServerFnComponent() -> impl IntoView {
+    let string_to_string = Resource::new(
+        || (),
+        |_| server_fn_to_string(String::from("I'm a String.")),
+    );
+    let u8_to_string = Resource::new(
+        move || (),
+        |_| async move { server_fn_to_string::<u8>(42).await },
+    );
+    view! {
+        <h3>Using generic function over display</h3>
+        <p>"This example demonstrates creating a generic function that takes any type that implements display that we've registered."</p>
+
+        <ErrorBoundary fallback=move |err|leptos::logging::log!("{err:?}")>
+
+        <Suspense >
+       <h4>Result 1</h4>
+           <p>
+            {
+                move || string_to_string.get()
+            }
+           </p>
+           <h4>Result 2</h4>
+           <p>
+           {
+               move || u8_to_string.get()
+           }
+           </p>
+        </Suspense>
+        </ErrorBoundary >
+
+
+    }
+}
+
+pub trait SomeTrait {
+    fn some_method() -> String;
+}
+
+pub struct AStruct;
+pub struct AStruct2;
+impl SomeTrait for AStruct {
+    fn some_method() -> String {
+        String::from("Hello world...")
+    }
+}
+impl SomeTrait for AStruct2 {
+    fn some_method() -> String {
+        String::from("HELLO WORLD")
+    }
+}
+
+#[server]
+#[register(<AStruct>,<AStruct2>)]
+pub async fn server_hello_world_generic<S: SomeTrait>(
+) -> Result<String, ServerFnError> {
+    Ok(S::some_method())
+}
+
+#[component]
+pub fn GenericHelloWorld() -> impl IntoView {
+    let s = RwSignal::new(String::new());
+    Effect::new(move |_| {
+        spawn_local(async move {
+            match server_hello_world_generic::<AStruct>().await {
+                Ok(hello) => s.set(hello),
+                Err(err) => leptos::logging::log!("{err:?}"),
+            }
+        })
+    });
+    let s2 = RwSignal::new(String::new());
+    Effect::new(move |_| {
+        spawn_local(async move {
+            match server_hello_world_generic::<AStruct2>().await {
+                Ok(hello) => s2.set(hello),
+                Err(err) => leptos::logging::log!("{err:?}"),
+            }
+        })
+    });
+    view! {
+        <h3>Using generic functions without generic inputs</h3>
+        <p>"This example demonstrates creating a generic server function that doesn't take any generic input."</p>
+        <h4>{"Results"}</h4>
+        <p>{ move || format!("With generic specified to {} we get {} from the server", std::any::type_name::<AStruct>(), s.get())}</p>
+        <p>{ move || format!("With generic specified to {} we get {} from the server", std::any::type_name::<AStruct2>(), s2.get())}</p>
+
+    }
+}
+
+#[derive(Clone)]
+struct SomeDefault;
+impl SomeTrait for SomeDefault {
+    fn some_method() -> String {
+        String::from("just a default hello world...")
+    }
+}
+
+#[server]
+#[register(<SomeDefault>)]
+pub async fn server_hello_world_generic_with_default<
+    S: SomeTrait = SomeDefault,
```

Comment:
Yes, I kind of tacked default on at the last minute as a "wouldn't this be cool" kind of feature. I came up against no default generic functions, I think I must have added it for server function structures but forget to try it against functions (I might have not gone ahead with it if I had lol). I'm not sure this feature makes sense for the reasons you stated i.e given that generics can't be default in server functions, and since the server function struct is where they'd have to be default in but that's not a given for any particular server function it is kind of confusing etc.